Java/DataStructures/LinkedList.java

```diff
@@ -169,7 +170,34 @@ public boolean contains(int value) {
         }
         return false;
     }
+    /*Mid of a linked list*/
+    public Node midNode(Node head) {
+        if (head == null || head.next == null)
+            return head;
+
+        Node slow = head, fast = head;
+        while (fast.next != null && fast.next.next != null) {
+            slow = slow.next;
+            fast = fast.next.next;
+        }
+
+        return slow;
+    }
+    
+    public Node midNode2(Node head) {
+        if (head == null || head.next == null)
+            return head;
+
+        Node slow = head, fast = head;
+        while (fast != null && fast.next != null) {
+            slow = slow.next;
+            fast = fast.next.next;
+        }
+
+        return slow;
+    }
 
+    
```

Comment:
Hi @gursparsh, 
The code seems good. 

Could you please call the function and add the sample input, output and time and space complexity as well?

If possible, could you add it for all the functions? (Let me know, if this is not convenient, I would take care of it later. )

Also, in the current version of this file, we have lines added due to merge conflicts, Line 1 for example.  This seems to have crept in the code, while we were organizing the JAVA folder for the better, for this repository.
Could you please remove such lines from the file, in the next review?

Also, let me know, if you would like to contribute to more issues / repository (Maintainer) in future. 

Thanks,
Sukriti

Reply:
Yeah sure! I'll do all the necessary changes and add time complexity for all the functions.
I would be happy to contribute to more issues !